### PR TITLE
clients(psi): add LH.Result object support to prepareLabData

### PR DESCRIPTION
**Pull request by exterkamp**
**Summary**
Allows prepareLabData to take in a `LH.Result` object as a param in addition to `string`.

**Related Issues/PRs**
depends on: #6424

### CODE REVIEW

**Comment by brendankenny** [comment]
I merged #6424, can you rebase off of master now?

File: lighthouse-core/test/report/html/renderer/psi-test.js [hunk 1]
```diff
@@ -20,8 +20,11 @@ const CriticalRequestChainRenderer =
     require('../../../../report/html/renderer/crc-details-renderer');
 
 const sampleResultsStr = fs.readFileSync(__dirname + '/../../../results/sample_v2.json', 'utf-8');
-const sampleResults = JSON.parse(sampleResultsStr)
-;
+const sampleResultsRoundtripStr = fs.readFileSync(
+  __dirname + '/../../../../../proto/sample_v2_round_trip.json',
```
Comment (brendankenny) [inline lighthouse-core/test/report/html/renderer/psi-test.js]: lol

File: lighthouse-core/test/report/html/renderer/psi-test.js [hunk 2]
```diff
@@ -59,6 +62,21 @@ describe('DOM', () => {
 
   describe('psi prepareLabData helpers', () => {
     describe('prepareLabData', () => {
+      it('succeeds with LHResult object (roundtrip) input', () => {
+        const roundTripLHResult = /** @type {LH.Result} */ JSON.parse(sampleResultsRoundtripStr);
```
Comment (brendankenny) [inline lighthouse-core/test/report/html/renderer/psi-test.js]: this seems good to have, though it makes #6405 a bit more work (e.g. this test will have to trigger a round trip result somehow). Can we prioritize that one soon so all the dependencies on whatever we come up with aren't too ossified?

File: lighthouse-core/test/report/html/renderer/psi-test.js (hunk 1 above)
Comment (exterkamp) [reply lighthouse-core/test/report/html/renderer/psi-test.js]: ../../../../.././lol.jpg/../../../../../../paths

File: lighthouse-core/test/report/html/renderer/psi-test.js [hunk 3]
```diff
@@ -59,6 +62,21 @@ describe('DOM', () => {
...
+        assert.ok(result.perfCategoryEl.outerHTML.length > 50000, 'perfCategory HTML is populated');
+        assert.ok(!result.perfCategoryEl.outerHTML.includes('lh-permalink'),
+            'PSI\'s perfCategory HTML doesn\'t include a lh-permalink element');
+        // Assume using default locale.
+        const titleEl = result.perfCategoryEl.querySelector('.lh-audit-group--metrics')
+          .querySelector('.lh-audit-group__header');
+        assert.equal(titleEl.textContent, Util.UIStrings.labDataTitle);
+      });
+
       it('reports expected data', () => {
```
Comment (paulirish) [inline lighthouse-core/test/report/html/renderer/psi-test.js]: > succeeds with stringified LHResult input

**Pull request merged**